examples/todoapp/lib/lib_permissions/data_sources/local/permissions_local_data_source.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:realm/realm.dart';
+
+import '../../../base/app/initialization/realm_instance.dart';
+import '../../models/permission_map_model.dart';
+import '../../models/permission_model.dart';
+
+class PermissionsLocalDataSource {
+  PermissionsLocalDataSource({required this.realmInstance});
+  final RealmInstance realmInstance;
+  static const String _permissionsKey = 'permissions';
+
+  void storePermissions(Map<String, bool> permissions) {
+    final oldPermission =
+        realmInstance.realm.find<PermissionModel>(_permissionsKey);
+
+    final RealmList<PermissionMap> permissionMap = RealmList<PermissionMap>(
+      permissions.entries.map(
+        (permission) => PermissionMap(permission.key, permission.value),
+      ),
+    );
+
+    /// Update permissions if they already exist
+    if (oldPermission != null) {
+      realmInstance.realm.write(() {
+        oldPermission.permissions.clear();
+        oldPermission.permissions.addAll(permissionMap);
+      });
+    }
+
+    /// Add permissions if they don't exist
+    else {
+      realmInstance.realm.write(() {
+        realmInstance.realm.add<PermissionModel>(
+          PermissionModel(_permissionsKey, permissions: permissionMap),
+        );
+      });
+    }
+  }
+
+  Map<String, bool> getPermissions() {
+    final permissions =
+        realmInstance.realm.find<PermissionModel>(_permissionsKey);
+
+    if (permissions == null) {
+      return {};
+    }
+
+    return {
+      for (var permission in permissions.permissions)
+        permission.permissionName: permission.isEnabled
+    };
```

Comment:
This code can be simplified without using imperative style.

Reply:
Analyzer recommends using a `for` loop here instead of a `Map.fromIterable`.
The only way that simplifies this is by using `Map.fromEntries`, but to me `for` is more readable

`return Map.fromEntries(
    permissions.permissions.map(
      (permission) => MapEntry(permission.permissionName, permission.isEnabled),
    ),
  );`